src/inspect_ai/model/_providers/vertex.py

```diff
@@ -336,10 +336,13 @@ async def content_part(content: Content | str) -> Part:
     elif isinstance(content, ContentImage):
         image_bytes, mime_type = await file_as_data(content.image)
         return Part.from_image(image=Image.from_bytes(data=image_bytes))
+    elif isinstance(content, ContentReasoning):
+        return Part.from_text(content.reasoning or NO_CONTENT)
     else:
         if isinstance(content, ContentAudio):
             file = content.audio
-        elif isinstance(content, ContentVideo):
+        else:
+            # it's ContentVideo
             file = content.video
         file_bytes, mime_type = await file_as_data(file)
         return Part.from_data(file_bytes, mime_type)
```

Comment:
This was a good one. It regressed when the reasoning stuff was added. Prior to that, the conditions were exhaustive. 

We could take another approach by leaving the `isinstance` check for video and having a final `else: assert False`. It's basically pick one type of crash or another. OR make the the result optional and make the callers deal. 

Thoughts?